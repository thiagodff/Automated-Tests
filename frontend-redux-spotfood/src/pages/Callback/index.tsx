import React, { useCallback, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { Spin } from 'antd';
import { signInCallback } from '../../store/modules/auth/actions';

import { Container } from './styles';

const Callback = () => {
  const dispatch = useDispatch();

  const checkCallbackPayload = useCallback(() => {
    dispatch(signInCallback());
  }, [dispatch]);

  useEffect(() => {
    checkCallbackPayload();
  }, [checkCallbackPayload, dispatch]);

  return (
    <Container>
      <Spin size="large" />
    </Container>
  );
}

export default Callback;