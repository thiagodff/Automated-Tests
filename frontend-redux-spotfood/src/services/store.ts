import { store } from '../store';

export function getSpotifyAccessToken() {
  return store.getState().auth.access_token;
}

export function getSpotifyAccessTokenType() {
  return store.getState().auth.token_type;
}