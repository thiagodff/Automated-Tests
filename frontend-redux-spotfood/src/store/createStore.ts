import { createStore, applyMiddleware, Middleware, Reducer } from 'redux';
import { AuthAction, AuthState } from './modules/auth/types';
import { PlaylistsAction, PlaylistsState } from './modules/playlists/types';
import { PersistState } from 'redux-persist';

export interface StoreState {
  _persist: PersistState;
  auth: AuthState;
  playlists: PlaylistsState;
}

export type StoreAction = AuthAction & PlaylistsAction;

export default (
  reducers: Reducer<StoreState, StoreAction>,
  middlewares: Middleware[]
) => {
  const enhancer = applyMiddleware(...middlewares);

  return createStore(reducers, enhancer);
};
