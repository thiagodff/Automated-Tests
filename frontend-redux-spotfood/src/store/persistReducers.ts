import storage from 'redux-persist/lib/storage';
import { persistReducer } from 'redux-persist';
import { Reducer } from 'redux';
import { StoreAction, StoreState } from './createStore';

export default (reducers: Reducer<StoreState, StoreAction>) => {
  const persistedReducer = persistReducer(
    {
      key: 'spotifood',
      storage,
      whitelist: ['auth', 'playlists'],
    },
    reducers
  );

  return persistedReducer;
};
