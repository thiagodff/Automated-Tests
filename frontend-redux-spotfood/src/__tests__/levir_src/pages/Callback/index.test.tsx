import React from 'react';
import { render } from '@testing-library/react';
import Callback from '../../../../pages/Callback';
import { Provider } from 'react-redux';
import { store } from '../../../../store'

describe('[PAGE]: Callback', () => {
  it('It should be render Callback page', () => {
    const { container } = render(
      <Provider store={store}>
        <Callback />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
