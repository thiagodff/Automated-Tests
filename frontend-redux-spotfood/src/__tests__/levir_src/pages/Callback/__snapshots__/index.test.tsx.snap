// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PAGE]: Callback It should be render Callback page 1`] = `
<div>
  <div
    class="sc-bdnylx dzlguk"
  >
    <div
      class="ant-spin ant-spin-lg ant-spin-spinning"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
</div>
`;
